iis/mymodule.cpp

```diff
@@ -755,17 +756,25 @@ CMyHttpModule::OnBeginRequest(
         goto Finished;
 	}
 
-    auto reportConfigurationError = [pConfig, pHttpContext] {
+    auto reportConfigurationError = [pConfig, pHttpContext, this] {
         pConfig->configLoadingFailed = true;
         pHttpContext->GetResponse()->SetStatus(500, "WAF internal error. Invalid configuration.");
         pHttpContext->SetRequestHandled();
+		LeaveCriticalSection(&m_csLock);
         return RQ_NOTIFICATION_FINISH_REQUEST;
     };
 
-    // If we previously failed to load the config, don't spam the event log by trying and failing again
+    // If we previously failed to load the config, try again if 10sec passed
     if (pConfig->configLoadingFailed)
     {
-        return reportConfigurationError();
+		if (difftime(curr_time, pConfig->configFailTime) < 10)
+			return reportConfigurationError();
+		else {
+			char* msg = "Recycling w3wp worker due to problems with loading modsec config";
+			WriteEventViewerLog(msg, EVENTLOG_ERROR_TYPE);
```

Comment:
Perhaps we can pass the literal message here in place to save one line of code, otherwise let's make it `const char*`